transport/src/main/java/io/netty/channel/socket/nio/NioSocketChannel.java

```diff
@@ -399,12 +399,12 @@ protected void doWrite(ChannelOutboundBuffer in) throws Exception {
                     // Zero length buffers are not added to nioBuffers by ChannelOutboundBuffer, so there is no need
                     // to check if the total size of all the buffers is non-zero.
                     ByteBuffer buffer = nioBuffers[0];
-                    int attemptedBytes = buffer.remaining();
                     final int localWrittenBytes = ch.write(buffer);
                     if (localWrittenBytes <= 0) {
                         incompleteWrite(true);
                         return;
                     }
+                    int attemptedBytes = buffer.remaining();
```

Comment:
This is not correct as the write will modify the amount of remaining bytes on success